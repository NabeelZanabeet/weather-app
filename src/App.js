import React, { Component } from 'react';
import {
  View
} from 'react-native';
import WeatherApp from './components/WeatherAppContainer';

export default class App extends Component {
  render() {
    return (
      <View>
        <WeatherApp />
      </View>
    );
  }
}
