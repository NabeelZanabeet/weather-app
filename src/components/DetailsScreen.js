import React from 'react';
import {
  View, Text, AppRegistry, Button, StyleSheet
} from 'react-native';
import DetailedWeatherCard from './DetailedWeatherCard';

const DetailsScreen = props => (
  <View>
    <Text style={styles.screen_title}>Details Screen</Text>
    <DetailedWeatherCard city={props.navigation.getParam('city')} />
    <Button
      title="Back to Home"
      onPress={() => props.navigation.navigate('Home')}
    />
  </View>
);

const styles = StyleSheet.create({
  screen_title: {
    margin: 10,
    fontSize: 20,
    color: 'black',
    textAlign: 'center'
  }
});

export default DetailsScreen;
AppRegistry.registerComponent('DetailsScreen', () => DetailsScreen);
