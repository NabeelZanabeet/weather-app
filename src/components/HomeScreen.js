/* eslint-disable no-console */
import React, { Component } from 'react';
import {
  View, Text, AppRegistry, FlatList, StyleSheet, TouchableHighlight
} from 'react-native';
import WeatherCard from './WeatherCard';
import PlusIcon from '../icons/PlusIcon';
import InputCityModal from './InputCityModal';

export default class HomeScreen extends Component {
  constructor() {
    super();
    this.state = {
      modalVisible: false
    };
  }

  setModalVisible = visible => {
    this.setState({ modalVisible: visible });
  }

  render() {
    const {
      cities, navigation, deleteCity, addCity
    } = this.props;
    const { modalVisible } = this.state;
    const { setModalVisible } = this;
    return (
      <View>
        <Text style={styles.screen_title}>Home Screen</Text>
        <TouchableHighlight
          style={styles.plus_icon}
          underlayColor="white"
          onPress={() => this.setModalVisible(true)}
        >
          <PlusIcon size={50} color="#900" />
        </TouchableHighlight>
        <InputCityModal
          cities={cities}
          modalVisible={modalVisible}
          setModalVisible={setModalVisible}
          addCity={addCity}
        />
        <FlatList
          style={styles.cities_flat_list}
          data={cities}
          keyExtractor={(item) => item.id.toString()}
          renderItem={
            ({ item }) => (
              <WeatherCard
                city={item}
                citiesCount={cities.length}
                navigation={navigation}
                deleteCity={deleteCity}
              />
            )
          }
        />
      </View>
    );
  }
}

const styles = StyleSheet.create({
  screen_title: {
    margin: 10,
    fontSize: 18,
    color: 'black',
    textAlign: 'center'
  },
  cities_flat_list: {
    height: 500
  },
  plus_icon: {
    justifyContent: 'center',
    alignSelf: 'center'
  }
});

AppRegistry.registerComponent('HomeScreen', () => HomeScreen);
