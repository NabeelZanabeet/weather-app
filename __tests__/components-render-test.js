// Note: test renderer must be required after react-native.
import 'react-native';
import React from 'react';
import renderer from 'react-test-renderer';
import { shallow } from 'enzyme';
import App from '../src/App';
import DetailsScreen from '../src/components/DetailsScreen';
import HomeScreen from '../src/components/HomeScreen';

it('DetailsScreen renders correctly', () => {
  renderer.create(<DetailsScreen />);
});

it('DetailsScreen should render', () => {
  const component = shallow(<DetailsScreen />);
  expect(component).toMatchSnapshot();
});
it('HomeScreen renders correctly', () => {
  renderer.create(<HomeScreen />);
});

it('HomeScreen should render', () => {
  const component = shallow(<HomeScreen />);
  expect(component).toMatchSnapshot();
});

it('App renders correctly', () => {
  renderer.create(<App />);
});

it('App should render', () => {
  const component = shallow(<App />);
  expect(component).toMatchSnapshot();
});
