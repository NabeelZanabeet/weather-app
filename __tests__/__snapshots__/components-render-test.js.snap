// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App should render 1`] = `
<View>
  <WeatherApp />
</View>
`;

exports[`DetailsScreen should render 1`] = `
<View>
  <Text>
    Details Screen
  </Text>
  <Button
    onPress={[Function]}
    title="Back to Home"
  />
</View>
`;

exports[`HomeScreen should render 1`] = `
<View>
  <Text>
    Home Screen
  </Text>
  <Button
    onPress={[Function]}
    title="Go to Details"
  />
</View>
`;
